#! /bin/sh -e
## 52_ip22-sercon.dpatch by Thiemo Seufer <seufer@csv.ica.uni-stuttgart.de>
##
## DP: Originally from Kaj-Michael Lang <milang@tal.org>
## DP: 
## DP: In ip22-setup.c the checks for serial/graphics console logic does
## DP: not check if ARCS console=g but the machine is using serial console, as
## DP: it does if no keyboard is attached.
## DP: 
## DP: This patch adds a check if ConsoleOut is serial. There might also be 
## DP: support for other graphics than Newport soon...

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

--- a/arch/mips/sgi-ip22/ip22-setup.c.orig	2005-02-17 01:53:53.000000000 +0100
+++ b/arch/mips/sgi-ip22/ip22-setup.c	2005-02-17 10:32:35.000000000 +0100
@@ -56,6 +56,7 @@ extern void ip22_time_init(void) __init;
 static int __init ip22_setup(void)
 {
 	char *ctype;
+	char *cserial;
 
 	board_be_init = ip22_be_init;
 	ip22_time_init();
@@ -81,9 +82,14 @@ static int __init ip22_setup(void)
 	/* ARCS console environment variable is set to "g?" for
 	 * graphics console, it is set to "d" for the first serial
 	 * line and "d2" for the second serial line.
+	 * 
+	 * Need to check if the case is 'g' but no keyboard:
+	 * (ConsoleIn/Out = serial)
 	 */
 	ctype = ArcGetEnvironmentVariable("console");
-	if (ctype && *ctype == 'd') {
+	cserial = ArcGetEnvironmentVariable("ConsoleOut");
+
+	if ((ctype && *ctype == 'd') || (cserial && *cserial == 's')) {
 		static char options[8];
 		char *baud = ArcGetEnvironmentVariable("dbaud");
 		if (baud)
@@ -91,7 +97,7 @@ static int __init ip22_setup(void)
 		add_preferred_console("ttyS", *(ctype + 1) == '2' ? 1 : 0,
 				      baud ? options : NULL);
 	} else if (!ctype || *ctype != 'g') {
-		/* Use ARC if we don't want serial ('d') or Newport ('g'). */
+		/* Use ARC if we don't want serial ('d') or graphics ('g'). */
 		prom_flags |= PROM_FLAG_USE_AS_CONSOLE;
 		add_preferred_console("arc", 0, NULL);
 	}
